### PR TITLE
Add parentheses to GETVAR macros

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- add parentheses to `GETVAR` and `ARR_SELECT` macros;
- remove superfluous parentheses in some scripts.

### CODE REVIEW

File: addons/main/script_macros.hpp [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #define GETGVAR(var1,var2) GETMVAR(GVAR(var1),var2)
 #define GETEGVAR(var1,var2,var3) GETMVAR(EGVAR(var1,var2),var3)
 
-#define ARR_SELECT(ARRAY,INDEX,DEFAULT) if (count ARRAY > INDEX) then {ARRAY select INDEX} else {DEFAULT}
+#define ARR_SELECT(ARRAY,INDEX,DEFAULT) (if (count ARRAY > INDEX) then {ARRAY select INDEX} else {DEFAULT})
```
Comment (commy2) [inline addons/main/script_macros.hpp]:
```sqf
(ARRAY param [INDEX, DEFAULT])
```

Comment (commy2) [reply addons/main/script_macros.hpp]: ^ Just an idea.

Comment (654wak654) [reply addons/main/script_macros.hpp]: Could also remove `ARR_SELECT` which only has 1 use:
https://github.com/acemod/ACE3/search?utf8=%E2%9C%93&q=ARR_SELECT&type=

Comment (Dystopian) [reply addons/main/script_macros.hpp]: what about BWC?

Comment (654wak654) [reply addons/main/script_macros.hpp]: ACE's macros aren't supposed to be used by others, specially not one this archaic.

Comment (jonpas) [reply addons/main/script_macros.hpp]: Actually they are: https://github.com/acemod/arma-project-template/blob/master/addons/main/script_macros.hpp

But this one can safely be removed as it plays no role anymore.

Comment (commy2) [reply addons/main/script_macros.hpp]: I mean, it can also stay as is. Doesn't have to be removed with this PR. I'm for merging this PR.

Comment (Dystopian) [reply addons/main/script_macros.hpp]: OK, will not update it

**Pull request merged**